http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/RunnableRequest.java

```diff
@@ -111,13 +112,33 @@ private URI endpoint() {
     private void handleFailure(Supplier<String> msg, Throwable cause) {
         log.error(msg.get(), cause);
         runAndLogError("Exception thrown from AsyncResponseHandler",
-            () -> context.handler().exceptionOccurred(cause));
+                       () -> context.handler().exceptionOccurred(modifyHighBurstTrafficException(cause)));
         if (channel != null) {
             runAndLogError("Unable to release channel back to the pool.",
-                () -> closeAndRelease(channel));
+                           () -> closeAndRelease(channel));
         }
     }
 
+    private Throwable modifyHighBurstTrafficException(Throwable originalCause) {
+        String originalMessage = originalCause.getMessage();
+        String newMessage = null;
+
+        if (originalCause instanceof TimeoutException &&
+            originalMessage.contains("Acquire operation took longer then configured maximum time")) {
+
+            newMessage = "Timed out waiting for a connection to be released to the pool. "
+                         + "Consider increasing your max connections or pool lease timeout.";
+
+        } else if (originalCause instanceof IllegalStateException &&
+                   originalMessage.contains("Too many outstanding acquire operations")) {
+            newMessage = "Request rate is too high. Please consider slowing down the rate of requests sent";
```

Comment:
Increasing max connections, decreasing pool lease timeout or increasing the outstanding queue size would also work.

Reply:
This error happens when the pending acquire queue is full.